### DISCUSSION TITLE
Paying fuel and hull repair

### DISCUSSION THREAD

**Discussion by ghost** [endless-sky/endless-sky#10881] [Balance, Mechanics, and Interactions]
Hi there. 
I'm doing a plugin trying to make some ways more difficult. One of this ways is about paying fuel/hull in human/hai space when the player wants (Or need). Searching the wiki know i can change the "spaceport" name for "port" and select one by one which services will be there.
But, i don't know how to get the actual fuel, the fuel capacity, and calculate the refill in a integer number. Like the "do you want to refill your anmo?" text.

¿Some ideas, please?.

**Comment by Hecter94** [comment]
I don't believe that's something a plugin would be able to do. I believe doing something like that would require engine changes.
Fuel/Hull aren't tangible items the same way ammo is.

If your goal is just to make the game harder, might I suggest simply using `on daily` to charge the player a set amount per day as a sort of "fuel/repair" tax?
If you wanted it to scale upwards as the player gets a larger ship, you could also use some autoconditions https://github.com/endless-sky/endless-sky/wiki/Player-Conditions#read-only to get the flagship model, attribute, category, etc., and trigger different missions based on that with different payment amounts.

**Reply by ziproot** [reply]
A word of warning: I tried something like this and ran into issues with missions triggering every day, as it blocked other missions from offering. However, with the new non-blocking tag, you can set those missions as non-blocking, allowing other spaceport missions to offer.

**Reply by ghost** [reply]
I thought something like you say, a daily tax. The other option for me is make 2 outfits: A fuel card and a insurance policy, both based in the ship's category, but for some reason the fuel card (Perhaps because my wife worked selling this, haha) seemed to me a bit ridiculous. You know, year 3013, and there's a "point travel card" yet.
The former approach was - for me - creating the outfits as support: If you want, you pay, and get benefits, if you don't want, you pay when you need.

Thanks for the replies. And sorry for my english, also, i believe that is a pain reading me xD.
